src/main/java/org/opensearch/neuralsearch/processor/TextImageEmbeddingProcessor.java

```diff
@@ -0,0 +1,227 @@
+    public void execute(final IngestDocument ingestDocument, final BiConsumer<IngestDocument, Exception> handler) {
+        try {
+            validateEmbeddingFieldsValue(ingestDocument);
+            Map<String, String> knnMap = buildMapWithKnnKeyAndOriginalValue(ingestDocument);
+            Map<String, String> inferenceMap = createInferences(knnMap);
+            if (inferenceMap.isEmpty()) {
+                handler.accept(ingestDocument, null);
+            } else {
+                mlCommonsClientAccessor.inferenceSentences(this.modelId, inferenceMap, ActionListener.wrap(vectors -> {
+                    setVectorFieldsToDocument(ingestDocument, vectors);
+                    handler.accept(ingestDocument, null);
+                }, e -> { handler.accept(null, e); }));
+            }
+        } catch (Exception e) {
+            handler.accept(null, e);
+        }
+
+    }
+
+    private void setVectorFieldsToDocument(IngestDocument ingestDocument, List<Float> vectors) {
+        Objects.requireNonNull(vectors, "embedding failed, inference returns null result!");
+        log.debug("Text embedding result fetched, starting build vector output!");
+        Map<String, Object> textEmbeddingResult = buildTextEmbeddingResult(this.embedding, vectors);
+        textEmbeddingResult.forEach(ingestDocument::setFieldValue);
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    private Map<String, String> createInferences(Map<String, String> knnKeyMap) {
+        Map<String, String> texts = new HashMap<>();
+        if (fieldMap.containsKey(TEXT_FIELD_NAME) && knnKeyMap.containsKey(fieldMap.get(TEXT_FIELD_NAME))) {
+            texts.put(INPUT_TEXT, knnKeyMap.get(fieldMap.get(TEXT_FIELD_NAME)));
+        }
+        if (fieldMap.containsKey(IMAGE_FIELD_NAME) && knnKeyMap.containsKey(fieldMap.get(IMAGE_FIELD_NAME))) {
+            texts.put(INPUT_IMAGE, knnKeyMap.get(fieldMap.get(IMAGE_FIELD_NAME)));
+        }
+        return texts;
+    }
+
+    @VisibleForTesting
+    Map<String, String> buildMapWithKnnKeyAndOriginalValue(IngestDocument ingestDocument) {
+        Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+        Map<String, String> mapWithKnnKeys = new LinkedHashMap<>();
+        for (Map.Entry<String, String> fieldMapEntry : fieldMap.entrySet()) {
+            String originalKey = fieldMapEntry.getValue(); // field from ingest document that we need to sent as model input, part of
+                                                           // processor definition
+
+            if (!sourceAndMetadataMap.containsKey(originalKey)) {
+                continue;
+            }
+            if (!(sourceAndMetadataMap.get(originalKey) instanceof String)) {
+                throw new IllegalArgumentException("Unsupported format of the field in the document, value must be a string");
+            }
+            if (((String) sourceAndMetadataMap.get(originalKey)).length() > MAX_CONTENT_LENGTH_IN_BYTES) {
```

Comment:
we should not add this check of max content length.

Reply:
I see, that is valid point, we may not know the size model will accept, so should not limit user. For AWS we may rely on network quotas which are even smaller for majority of instances https://docs.aws.amazon.com/opensearch-service/latest/developerguide/limits.html

Reply:
if this is a mandatory check we can add a parameter in the processor which defines the max size and we can put the default value as the current value which is provided in this PR.

Reply:
it's not a must for now, I removed the check, we can add processor param or other way to configure that in next versions